streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -88,30 +93,50 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry,
         final ProcessorRecordContext current = context.recordContext();
         try {
             context.setRecordContext(entry.entry().context());
+
+            final byte[] newValue = entry.newValue();
             if (flushListener != null) {
-                V oldValue = null;
-                if (sendOldValues) {
-                    final byte[] oldBytesValue = underlying.get(entry.key());
-                    oldValue = oldBytesValue == null ? null : serdes.valueFrom(oldBytesValue);
+                final byte[] oldRawValue;
```

Comment:
I personally prefer using `final` -- immutability is a nice property. Not sure what @guozhangwang @vvcephei think.

Reply:
Yeah that's fine with me, just a minor point

Reply:
Also prefer to have final variables, and actually I prefer to have "complete" instead of "one-sided" conditionals, too. (See https://github.com/kubernetes/kubernetes/blob/ec2e767e59395376fa191d7c56a74f53936b7653/pkg/controller/volume/persistentvolume/pv_controller.go#L55 for a succinct justification of this style)

If you want to tighten it up, a ternary operator would do it.